### PR TITLE
Fix: `appliedMigrations` throws error on new install

### PR DESCRIPTION
**Pull request by michaelkirk-signal**
Before setup, the grdb_migrations table doesn't exist, so calling
`appliedMigrations` would throw an error.

To avoid this, if the table to track migrations doesn't exist yet, we can be
sure no migrations have been run, and we retun the empty set.

FIXES https://github.com/groue/GRDB.swift/issues/652

### Pull Request Checklist

- [x] This pull request is submitted against the `development` branch.
- [n/a] Inline documentation has been updated.
- [n/a] README.md or another dedicated guide has been updated.
- [x] Changes are tested.

### CODE REVIEW

File: GRDB/Migration/DatabaseMigrator.swift [hunk 1]
```diff
@@ -213,7 +213,17 @@ public struct DatabaseMigrator {
     
     /// Returns the set of applied migration identifiers.
     public func appliedMigrations(in reader: DatabaseReader) throws -> Set<String> {
-        return try reader.read { try appliedIdentifiers($0) }
+        return try reader.read { db in
+            guard let doesMigrationsTableExist = try Bool.fetchOne(db, sql: "SELECT EXISTS (SELECT 1 FROM sqlite_master WHERE type='table' AND name='grdb_migrations')") else {
```
Comment (groue) [inline GRDB/Migration/DatabaseMigrator.swift]: I can't help thinking that `appliedMigrations(in:)` is fixed, but the private `appliedIdentifiers` method is still sensitive to the lack of the grdb_migrations tables, for no good reason.

Maybe a better fix would be to fix the root `appliedIdentifiers` method itself.

```swift
public func appliedMigrations(in reader: DatabaseReader) throws -> Set<String> {
    return try reader.read(appliedIdentifiers)
}

private func appliedIdentifiers(_ db: Database) throws -> Set<String> {
    let tableExist = try Bool.fetchOne(db, sql: """
        SELECT EXISTS (SELECT 1 FROM sqlite_master WHERE type='table' AND name='grdb_migrations')
        """)!
    guard tableExist else {
        return []
    }
    return try Set(String.fetchAll(db, sql: "SELECT identifier FROM grdb_migrations"))
}
```

Also note how the force unwrap of the Bool is just enough: SQLite will never return zero row or NULL.

File: Tests/GRDBTests/DatabaseMigratorTests.swift [hunk 2]
```diff
@@ -236,6 +236,31 @@ class DatabaseMigratorTests : GRDBTestCase {
             }
         }
     }
+
+    func testAppliedMigrations() throws {
```
Comment (groue) [inline Tests/GRDBTests/DatabaseMigratorTests.swift]: Amazing, there was no plain test of this method yet 🤣

Thanks 👍

File: GRDB/Migration/DatabaseMigrator.swift (hunk 1 above)
Comment (michaelkirk-signal) [reply GRDB/Migration/DatabaseMigrator.swift]: I've pushed up a new commit which does this.

**Comment by groue** [comment]
> error: unexpected service error: The Xcode build system has crashed. Please close and reopen your workspace.

Travis is sooo flaky. Let's wait a little more…

**Comment by groue** [comment]
Yes! I'll ship 4.6.2 shortly. Thank you Michael!

**Comment by groue** [comment]
Done! CocoaPods will be updated when `pod trunk push` has ended.

**Pull request merged**